freud/common.pyx

```diff
@@ -42,91 +39,41 @@ cdef class Compute:
     """
 
     def __cinit__(self):
-        self._called_compute = {"compute": False}
-
-    def _set_compute_flag(self, flag_name):
-        self._called_compute[flag_name] = False
+        self._called_compute = False
+
+    def __getattribute__(self, attr):
```

Comment:
I'm not sure if I like removing the decorators and relying on "magic names," since it's less transparent for new developers. But I don't have strong feelings on it.

Reply:
I had the same thought for a bit. But I do like that it provides a tiny bit of extra enforcement for our API standards. It also significantly simplifies things given that aside from properties all the other methods are very standard (`compute`, `reset`, and `plot` are well-defined standards in the context of the new freud API IMO).

Reply:
Good enough justification -- resolving this.